### PR TITLE
Allow per-clip and per-track colors

### PR DESCRIPTION
**Pull request by jminor**
This begins to address #11

For now, this stores colors in a metadata field: `"raven": {"color": "CYAN"}`

Here's what it looks like.
Screenshot 2023-05-30 at 9 54 41 PM

### CODE REVIEW

File: editing.cpp [hunk 1]
```diff
@@ -220,3 +221,34 @@ void FlattenTrackDown() {
...
 }
+
+std::string GetItemColor(otio::Item* item)
+{
+    std::string item_color = "";
+
+    if (item->metadata().has_key("raven") &&
+        item->metadata()["raven"].type() == typeid(otio::AnyDictionary))
+    {
+        auto raven_md = otio::any_cast<otio::AnyDictionary>(item->metadata()["raven"]);
```
Comment (jminor) [inline editing.cpp:232]: Is there a more elegant way to get/set a field within a nested metadata dictionary? This feels very awkward compared to the Python equivalent.

Comment (meshula) [reply editing.cpp:232]: AFAIK, what you are doing is as compact as it gets. I think we'd have to do some work with template and operator magic to create a kind of DSL that would let you index deeply into nested dictionaries.

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
Our Marker schema has a color property (with the enumeration you're using here). I'm wondering if it's time to talk about migrating the property higher in the schema, so that it becomes available to any composable element?

**Comment by timlehr** [comment]
Very cool! I wonder if this should be the beginning of discussion about a custom schema for Raven specific metadata.

**Comment by jminor** [comment]
Discussion about OTIO schema for per-clip colors can continue here: https://github.com/AcademySoftwareFoundation/OpenTimelineIO/issues/1614

**Pull request merged**